Signal/src/ViewControllers/OWSSoundSettingsViewController.m

```diff
@@ -0,0 +1,165 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+#import "OWSSoundSettingsViewController.h"
+#import <AVFoundation/AVFoundation.h>
+#import <SignalMessaging/OWSSounds.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface OWSSoundSettingsViewController ()
+
+@property (nonatomic) BOOL isDirty;
+
+@property (nonatomic) OWSSound currentSound;
+
+@property (nonatomic) AVAudioPlayer *audioPlayer;
+
+@end
+
+#pragma mark -
+
+@implementation OWSSoundSettingsViewController
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+
+    switch (self.soundType) {
+        case OWSSoundType_Notification:
+            [self setTitle:NSLocalizedString(@"SETTINGS_ITEM_NOTIFICATION_SOUND",
+                                             @"Label for settings view that allows user to change the notification sound.")];
+            self.currentSound
+            = (self.thread ? [OWSSounds notificationSoundForThread:self.thread] : [OWSSounds globalNotificationSound]);
+            break;
+        case OWSSoundType_Ringtone:
+            [self setTitle:NSLocalizedString(@"SETTINGS_ITEM_RINGTONE_SOUND",
+                                             @"Label for settings view that allows user to change the ringtone sound.")];
+            self.currentSound
+            = (self.thread ? [OWSSounds ringtoneSoundForThread:self.thread] : [OWSSounds globalRingtoneSound]);
+            break;
+    }
+
+    [self updateTableContents];
+    [self updateNavigationItems];
+}
+
+- (void)viewDidAppear:(BOOL)animated
+{
+    [self updateTableContents];
+}
+
+- (void)updateNavigationItems
+{
+    self.navigationItem.leftBarButtonItem =
+        [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemCancel
+                                                      target:self
+                                                      action:@selector(cancelWasPressed:)];
```

Comment:
Should we use "cancel" or "stop"?  We tend to use "stop" throughout our app.

Reply:
IMO I agree with "Cancel", though I don't feel strongly.